Feat(#1146): UI - Landing Page

akhilmhdh opened the pull request:
<!--
Follow the steps to create the PR:

Subject: "feat/fix/docs(#issue_id): <PR Subject>"
Assignees: "One Platform Developers"
Projects: "One Platform Development"
-->

# Closes/Fixes/Resolves

### <!-- Any of the keywords: Closes/Fixes/Resolves followed by #issue_id (should be separated by a space only) -->

Closes 1146.

# Explain the feature/fix

1. Migrated to react
2. Setup eslint and prettier
3.  Completed UI landing page base
4. Added infinite scrolling

## Does this PR introduce a breaking change

Technically yes practically no

## Screenshots

<details>
<summary>View Screenshots</summary>

| Page | View |
|---|---|
| Home Page | <img width="1440" alt="Screenshot 2021-06-22 at 1 50 11 PM" src="https://user-images.githubusercontent.com/31166322/122890680-887b4300-d361-11eb-92d0-8ba2b4cf98b6.png"> |
| Homepage sticky layout on scroll | <img width="1440" alt="Screenshot 2021-06-22 at 1 49 51 PM" src="https://user-images.githubusercontent.com/31166322/122891607-456d9f80-d362-11eb-8ba1-f60d6fd067f9.png"> |
| New Idea Modal | <img width="1440" alt="Screenshot 2021-06-22 at 2 02 29 PM" src="https://user-images.githubusercontent.com/31166322/122891995-a1382880-d362-11eb-9da9-ee45230f02c2.png"> |

</details>

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [ ] Does the change have appropriate unit tests?
- [ ] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [ ] Was this feature demo'd and the design review approved?

deshmukhmayur reviewed (changes requested):
Also, can you add screenshots too the PR description. Even if it's not completely done, just a screenshot of how it looks so far.

riginoommen commented on package.json:
```diff
@@ -1,38 +1,67 @@
+    "name": "idea-hub",
+    "version": "0.1.0",
+    "private": true,
+    "dependencies": {
```
Include author information

akhilmhdh commented:
@deshmukhmayur Yah sure i'll add screenshots. I just kinda copied readme from the video library SPA and present one of vue. Ill push with these changes soon

deshmukhmayur commented:
> @deshmukhmayur Yah sure i'll add screenshots. I just kinda copied readme from the video library SPA and present one of vue. Ill push with these changes soon

Got it. @akhilmhdh If this is a work in progress PR, then you can convert it into a Draft PR by clicking on the link under the Reviewers section in the right sidebar^

akhilmhdh commented:
@deshmukhmayur  added screenshots on top.

deshmukhmayur commented on src/App.scss:
```diff
@@ -8,6 +8,13 @@ body,
   width: 100%;
   padding: 0;
   margin: 0;
+  height: 100vh;
```
Any specific reason why the page has a fixed height?

akhilmhdh replied:
Ohhh that's a typo it can be height 100% too. But I need a dimension regarding height there so that sticky will know its parent height.

And yes i can remove that overflow. But just specified there to avoid browser automatic decision.

akhilmhdh commented:
@deshmukhmayur  Fixed the sticky issue. You were right  there was some issue in a component. 

The issue was inside patternfly page component. Inside page component there exist a main component. Main component had overflow as auto so to make sticky inside that height must be said. Removed that overflow and its working fine now. Also simplified layout.

## Screenshot

<img width="1440" alt="Screenshot 2021-06-23 at 8 01 38 PM" src="https://user-images.githubusercontent.com/31166322/123116433-88ad3880-d45e-11eb-8d9f-9e0d222a8212.png">

riginoommen commented on README.md:
```diff
@@ -1,40 +1,33 @@
-# Idea Hub Spa
+# Idea Hub SPA for One-Platform
```
```suggestion
# Idea Hub SPA for One Platform
```

riginoommen commented on README.md:
```diff
@@ -1,40 +1,33 @@
+# Idea Hub SPA for One-Platform
 
-A hub to share ideas and innovations
+Idea Hub is a hub to share ideas and innovations, with feedbacks from other users. This repository is part of [One Platform](https://github.com/1-Platform/one-platform).
```
```suggestion
Idea Hub is a place to share ideas and innovations, with feedbacks from other users. This repository is part of [One Platform](https://github.com/1-Platform/one-platform).
```

riginoommen commented on README.md:
````diff
@@ -1,40 +1,33 @@
-```
-npm install
-```
+## Prerequisite
````
```suggestion
## Prerequisites
```

riginoommen commented on README.md:
````diff
@@ -1,40 +1,33 @@
-```
-npm run serve
-```
+1. Node
+2. Npm
````
Please do mention the versions supported

gautamkrishnar commented on src/App.test.tsx:
```diff
@@ -0,0 +1,9 @@
+
+test('renders learn react link', () => {
+  render(<App />);
+  const linkElement = screen.getByText(/React + Typescript/i);
```
Plese remve this test, can you please add some basic tests.

gautamkrishnar reviewed (changes requested):
Add basic snapshot test with that mock data for App.tsx and other components. You can use https://mswjs.io/ for mocking the actual data.

gautamkrishnar reviewed (dismissed):
LGTM 👍🏻

riginoommen commented on README.md:
````diff
@@ -1,40 +1,33 @@
-```
-npm run serve
-```
+1. node[v14]
+2. npm[v6]
````
```suggestion
1. Nodejs v14.17.1
2. NPM v6.14.10
```

riginoommen commented on README.md:
````diff
@@ -1,40 +1,33 @@
-```
-npm run test:unit
-```
+1. `git clone && cd <into the repo>`
````
```suggestion
1. Clone athe repository and switch into directory
`git clone https://github.com/1-Platform/idea-hub.git && cd idea-hub`
```

riginoommen commented on README.md:
````diff
@@ -1,40 +1,33 @@
-npm run test:unit
-```
+1. `git clone && cd <into the repo>`
+2. `npm install`
````
```suggestion
2.  Install Dependencies
`npm install`
```

ghanlohar reviewed (approved):
LGTM
